elastic-container.sh

```diff
@@ -15,92 +15,185 @@
-docker.elastic.co/kibana/kibana:${STACK_VERSION}
-
-# Start the Elastic Fleet Server container
-docker run -d --network elastic --rm --name fleet-server -p 8220:8220 \
--e "KIBANA_HOST=${KIBANA_URL}" \
--e "KIBANA_USERNAME=elastic" \
--e "KIBANA_PASSWORD=${KIBANA_PASSWORD}" \
--e "ELASTICSEARCH_HOSTS=${ELASTICSEARCH_URL}" \
--e "ELASTICSEARCH_USERNAME=elastic" \
--e "ELASTICSEARCH_PASSWORD=${ELASTIC_PASSWORD}" \
--e "KIBANA_FLEET_SETUP=1" \
--e "FLEET_SERVER_ENABLE=1" \
--e "FLEET_SERVER_INSECURE_HTTP=1" \
-docker.elastic.co/beats/elastic-agent:${STACK_VERSION}
-echo
-echo "Browse to http://localhost:5601"
-echo "Username: elastic"
-echo "Passphrase: ${ELASTIC_PASSWORD}"
-
-# Stop and remove the Elastic containers and network
-else
-if [ $1 == stop ] 2> /dev/null
-then
-echo "#####"
-echo "Stopping and removing all Elastic Stack components."
-echo "#####"
-docker stop fleet-server 2> /dev/null
-docker stop kibana 2> /dev/null
-docker stop elasticsearch 2> /dev/null
-docker network rm elastic 2> /dev/null
-
-# Restart the Elastic containers
-else
-if [ $1 == restart ] 2> /dev/null
-then
-echo "#####"
-echo "Restarting all Elastic Stack components."
-echo "#####"
-docker restart elasticsearch 2> /dev/null
-docker restart kibana 2> /dev/null
-docker restart fleet-server 2> /dev/null
-
-# Get the status of the Elastic containers
-else
-if [ $1 == status ] 2> /dev/null
-then
-docker ps -f "name=kibana" -f "name=elasticsearch" -f "name=fleet-server" --format "table {{.Names}}: {{.Status}}"
+  actions:
+    stage     downloads all necessary images to local storage
+    start     creates network and configures containers to run
+    stop      stops the containers created and removes the network
+    restart   simply restarts all the stack containers
+    status    check the status of the stack containers
+    help      print this message
+
+  flags:
+    -v        enable verbose output
+EOF
+}
+
+configure_kbn() {
+  MAXTRIES=15
+  i=${MAXTRIES}
+
+  while [ $i -gt 0 ]; do
+    STATUS=$(curl -I "${LOCAL_KBN_URL}" 2>&3 | head -n 1 | cut -d$' ' -f2)
+
+    echo
+    echo "Attempting to enable the Detection Engine and Prebuilt-Detection Rules"
+
+    if [ "${STATUS}" == "302" ]; then
+      echo
+      echo "Kibana is up. Proceeding"
+      output=$(curl --silent "${HEADERS[@]}" --user "${ELASTIC_USERNAME}:${ELASTIC_PASSWORD}" -XPOST "${LOCAL_KBN_URL}/api/detection_engine/index" 2>&3)
+      [[ $output =~ '"acknowledged":true' ]] || (
+        echo "Detection Engine setup failed :-("
+        exit 1
+      )
+
+      echo "Detection engine enabled. Installing prepackaged rules."
+      curl --silent "${HEADERS[@]}" --user "${ELASTIC_USERNAME}:${ELASTIC_PASSWORD}" -XPUT "${LOCAL_KBN_URL}/api/detection_engine/rules/prepackaged" 1>&3 2>&3
+
+      echo
+      echo "Prebuilt Detections Enabled!"
+      break
+    else
+      echo
+      echo "Kibana still loading. Trying again in 40 seconds"
+    fi
+
+    sleep 40
```

Comment:
Is this trying every 40 seconds? This doesn't take this long, can we change this to a 10-second retry?

Fix [here](https://github.com/peasead/elastic-container/pull/1/files#r794911672) too.

Reply:
It takes exactly 40 seconds so it never has to try again unless something with the container setup goes horribly wrong